spec/requests/api/v1/provider_spec.rb

```diff
@@ -290,6 +290,62 @@ def get_provider_codes_from_body(body)
           expect(returned_provider_codes.size).to eq 0
         end
       end
+
+      context 'a single provider with multiple published enrichments' do
+        let!(:provider) do
+          create(:provider,
+                site_count: 0,
+                region_code: 6,
+                created_at: '2019-01-21T15:44:32.1605',
+                updated_at: '2019-02-04T14:20:58.617561',
+                enrichments: [new_published_enrichment, old_published_enrichment])
+        end
+        let!(:site1) do
+          create(:site,
+                created_at: "2019-02-04T17:20:02.194607",
+                updated_at: "2019-02-04T17:20:02.194608",
+                provider: provider)
+        end
+        let!(:site2) do
+          create(:site,
+                created_at: "2019-02-04T17:20:02.19461",
+                updated_at: "2019-02-04T17:20:02.194611",
+                provider: provider)
+        end
+        let(:new_published_enrichment) do
+          build(:provider_enrichment,
+                address1: 'enrichment1 address1',
+                address2: 'enrichment1 address2',
+                address3: 'enrichment1 address3',
+                address4: 'enrichment1 address4',
+                postcode: 'enrichment1 postcode',
+                region_code: 6,
+                created_at: "2019-02-05T10:32:14.456414",
+                updated_at: "2019-02-05T10:32:22.95923",
+                status: 1)
+        end
+        let(:old_published_enrichment) do
+          build(:provider_enrichment,
+                region_code: 6,
+                created_at: "2018-09-26T15:25:33.191305",
```

Comment:
Might be easier to read using a date object i.e. `Date.yesterday`

Reply:
I was using the exact dates that was showing the bug.